dnstable/query.c

```diff
@@ -528,6 +529,123 @@ increment_key(ubuf *key, size_t pos)
 	return (dnstable_res_failure);
 }
 
+#define SL_MATCHES	0		/* Label matches. */
+#define SL_NOMATCH	1		/* Not a match. */
+#define SL_FAILURE	2		/* Error detected. */
+
+/*
+ * Used with a "single label" wildcard search to determine the action to take
+ * for a given matching key.
+ * Either:
+ *  + Matches  -- One and only one extra label.
+ *  + No match -- Restart search.
+ *  + Failure  -- Some error.
+ *
+ * Return-value indicates the action to take.
+ */
+static int
```

Comment:
Done; used ```enum```.
The failure handling mimics what is done already in ```query_iter_next_ip()```.
In addition, it is more efficient to return immediately upon failure, rather than fall-through and wait for a subsequent ```mtbl_``` operation to fail. In two of the three current calls, there would be redundant processing that happens.